tests/rpc/rpc_01-system_test.go

```diff
@@ -0,0 +1,182 @@
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSystemRPC(t *testing.T) {
+	if GOSSAMER_INTEGRATION_TEST_MODE != rpcSuite {
+		_, _ = fmt.Fprintln(os.Stdout, "Going to skip RPC suite tests")
+		return
+	}
+
+	testsCases := []struct {
+		description string
+		method      string
+		expected    interface{}
+		skip        bool
+	}{
+		{ //TODO
+			description: "test system_name",
+			method:      "system_name",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_version",
+			method:      "system_version",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_chain",
+			method:      "system_chain",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_properties",
+			method:      "system_properties",
+			skip:        true,
+		},
+		{
+			description: "test system_health",
+			method:      "system_health",
+			expected: modules.SystemHealthResponse{
+				Health: common.Health{
+					Peers:           2,
+					IsSyncing:       false,
+					ShouldHavePeers: true,
+				},
+			},
+			skip: false,
+		},
+		{
+			description: "test system_peers",
+			method:      "system_peers",
+			expected: modules.SystemPeersResponse{
+				Peers: []common.PeerInfo{},
+			},
+			skip: false,
+		},
+		{
+			description: "test system_network_state",
+			method:      "system_networkState",
+			expected: modules.SystemNetworkStateResponse{
+				NetworkState: common.NetworkState{
+					PeerID: "",
+				},
+			},
+			skip: false,
+		},
+		{ //TODO
+			description: "test system_addReservedPeer",
+			method:      "system_addReservedPeer",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_removeReservedPeer",
+			method:      "system_removeReservedPeer",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_nodeRoles",
+			method:      "system_nodeRoles",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_accountNextIndex",
+			method:      "system_accountNextIndex",
+			skip:        true,
+		},
+	}
+
+	t.Log("going to start gossamer")
+
+	localPidList, err := StartNodes(t, make([]*exec.Cmd, 1))
+
+	//use only first server for tests
+	require.Nil(t, err)
+
+	time.Sleep(time.Second) // give server a second to start
+
+	for _, test := range testsCases {
+		t.Run(test.description, func(t *testing.T) {
+			if test.skip {
+				t.Skip("RPC endpoint not yet implemented")
+				return
+			}
+
+			respBody, err := PostRPC(t, test.method, "http://"+GOSSAMER_NODE_HOST+":"+currentPort, "{}")
+			require.Nil(t, err)
+
+			target := DecodeRPC(t, respBody, test.method)
+
+			require.NotNil(t, target)
+
+			t.Log("Will start assertion for ", "target", target, "type", reflect.TypeOf(target))
+
+			switch v := target.(type) {
+
+			case *modules.SystemHealthResponse:
+				t.Log("Will assert SystemHealthResponse", "target", target)
+
+				require.Equal(t, test.expected.(modules.SystemHealthResponse).Health.IsSyncing, v.Health.IsSyncing)
```

Comment:
this line could potentially panic - for example, say system_networkState returns a `SystemHealthResponse` for some reason (there are probably better examples in the other modules but this is these are the only ones implemented so far) 
then, test.expected is `SystemNetworkStateResponse`, not `SystemHealthResponse`. then, the type conversion `test.expected.(modules.SystemHealthResponse)` will panic.

Reply:
But it should not return that, I mean, why would the system_networkState return a type that belongs to the other endpoint ? that should be captured at the unit test level. This are integration tests.

Reply:
We are assuming here, that the minimum tests have been already checked by proper unit testing. Again, if this is happening at the integration tests and we have a panic, that is a good thing to happen. At least we will know right away something is really broken.

Reply:
okay I just think it's good practice to always do an `a, ok := b.(mytype)` to avoid potential panics, it also seems to make more sense to switch on the test description or method name instead of the expected type, but if you prefer it this way

Reply:
sure, raised #815 to tackle this!